torch/csrc/autograd/variable.h

```diff
@@ -218,14 +220,17 @@ inline std::shared_ptr<Function>& Variable::grad_fn() {
   return get()->get_grad_fn();
 };
 inline void Variable::set_history(VarFlags flags, int output_nr, std::shared_ptr<Function> grad_fn) {
+  assert(!get()->_grad_fn && "set_history can only be called on new Variables");
```

Comment:
`TORCH_ASSERT`. This would never get triggered because non-debug builds of Python build extensions with `-DNDEBUG`